app/models/manageiq/providers/ansible_tower/shared/inventory/parser/automation_manager.rb

```diff
@@ -74,27 +74,32 @@ def configuration_script_sources
     end
   end
 
+  def miq_credential_types(provider_module)
```

Comment:
This isn't good.

It says that the first time you call this method with a provider_module, it will cache the `miq_credential_types` with that provider module.  If you call again with a different provider module, it will give you the old value.

Refactoring this out is probably creating too much of a problem.  You're going to keep digging down until you have a solution that's more complicated than the problem it's trying to solve.

You should probably just inline this again in the `credentials` method, and don't cache the result in `miq_credential_types`.

If we find that's causing some performance issues one day, we can revisit this.

Reply:
This parser is an instance and shouldn't be used for other `provider_module`